mapie/tests/test_mondrian.py

```diff
@@ -0,0 +1,434 @@
+    mondrian.fit(x, y, groups=groups)
+    groups = np.array([99] * len(x))
+    with pytest.raises(ValueError, match=r".*There is at least one new*"):
+        mondrian.predict(x, groups=groups, alpha=.2)
+
+
+def test_groups_and_x_have_same_length_in_predict():
+    """
+    Test that groups and x have the same length in predict"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x))
+    mondrian.fit(x, y, groups=groups)
+    groups = np.random.choice(10, len(x) - 1)
+    with pytest.raises(ValueError, match=r".*The number of individuals in*"):
+        mondrian.predict(x, groups=groups, alpha=.2)
+
+
+def test_alpha_none_return_one_element():
+    """
+    Test that if alpha is None, the output is a single element"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x))
+    mondrian.fit(x, y, groups=groups)
+    preds = mondrian.predict(x, groups=groups)
+    assert len(preds) == len(x)
+
+
+def test_groups_is_list_ok():
+    """
+    Test that the groups can be a list"""
+    x, y = TOY_DATASETS["classification"]
+    ml_model = ML_MODELS["classification"]
+    model = clone(ml_model)
+    model.fit(x, y)
+    mondrian = MondrianCP(
+        mapie_estimator=MapieClassifier(estimator=model, cv="prefit")
+    )
+    groups = np.random.choice(10, len(x)).tolist()
+    mondrian.fit(x, y, groups=groups)
+    mondrian.predict(x, groups=groups, alpha=.2)
+
+
+@pytest.mark.parametrize("mapie_estimator_name", VALID_MAPIE_ESTIMATORS_NAMES)
+def test_same_results_if_only_one_group(mapie_estimator_name):
+    """
+    Test that the results are the same if there is only one group"""
+    task_dict = VALID_MAPIE_ESTIMATORS[mapie_estimator_name]
+    mapie_estimator = task_dict["estimator"]
+    mapie_kwargs = task_dict["kwargs"]
+    task = task_dict["task"]
+    x, y = TOY_DATASETS[task]
+    y = np.abs(y)
+    ml_model = ML_MODELS[task]
+    groups = [0] * len(x)
+    model = clone(ml_model)
+    model.fit(x, y)
+    mapie_inst_mondrian = deepcopy(mapie_estimator)
+    mapie_classic_inst = deepcopy(mapie_estimator)
+    mondrian_cp = MondrianCP(
+        mapie_estimator=mapie_inst_mondrian(
+            estimator=model, cv="prefit", random_state=0, **mapie_kwargs
+        )
+    )
+    mapie_classic = mapie_classic_inst(
+        estimator=model, cv="prefit", random_state=0, **mapie_kwargs,
+    )
+    mondrian_cp.fit(x, y, groups=groups)
+    mapie_classic.fit(x, y)
+    mondrian_pred = mondrian_cp.predict(x, groups=groups, alpha=.2)
+    classic_pred = mapie_classic.predict(x, alpha=.2)
```

Comment:
It might be interesting to run this test with different alpha values, to avoid the side effect of data size.

Reply:
done